src/meta/src/hummock/compaction/level_selector.rs

```diff
@@ -220,6 +220,20 @@ impl DynamicLevelSelector {
         ctx.score_levels.sort_by(|a, b| b.0.cmp(&a.0));
         ctx
     }
+
+    fn adjust_target_file_size(&self, mut ret: SearchResult, base_level: usize) -> SearchResult {
```

Comment:
1. i think need more notes for this algorithm (e.g. simply explain about reduce write amplification)
2. IMO, the parameter name call `ret` is not to make sense

Reply:
How about renaming `SearchResult` to `CompactionInput` or `CompactionContext`?